src/ape/contracts/base.py

```diff
@@ -621,6 +621,11 @@ def poll_logs(
         required_confirmations = (
             required_confirmations or self.provider.network.required_confirmations
         )
+        
+        height = max(self.chain_manager.blocks.height - required_confirmations, 0)
+        if start_block and height > 0 and start_block < height:
+            yield from self.range(start_block, height)
```

Comment:
this is a nice optimization! but i think this deserves a comment because at first i was like:

`tHeSe bLoCkS aRe aLrEaDy hAnDleD beLoW!!!1`

but after reading the issue, i understand why we pre-handle them here.

Reply:
done